---
title: "RNA-Seq data analysis"
output: html_notebook
---

Introduction
============

We're going to compare the expression levels of all genes in two conditionsl, Ish and T47D.  We have 5 replicates for each condition.


Setup
=====

```{r message=FALSE}
library(tidyverse)
```


Prepare the data
================

Read the TSV
------------

```{r "Read Data TSV", message=FALSE}
read_tsv(
  "rnaseq_counts.txt",
  guess_max=1000000
) -> data

head(data)
```
Restructuring
-------------

We're going to split apart the annotation and the data to save space.

```{r}
data %>%
  select(Probe:Distance) -> data.annotation
```

We can then remove the annotation from the data and restructure it.


```{r}
data %>%
  select(-(Chromosome:Distance)) %>%
  pivot_longer(
    cols=-Probe,
    names_to="sample",
    values_to="count"
  ) %>%
  separate(
    sample,
    into=c("condition","replicate"),
    sep="_",
    remove = FALSE
  ) -> data

head(data)
```
Let's remove any genes which are unmeasured in all samples.

```{r}
data %>%
  group_by(Probe) %>%
  summarise(
    max = max(count)
  ) %>%
  ungroup() %>%
  filter(max==0) %>%
  pull(Probe) -> unmeasured.probes


data %>%
  filter(! Probe %in% unmeasured.probes ) -> data
```

Normalisation
-------------












