---
title: "RNA-Seq data analysis"
output: html_notebook
---

Introduction
============

We're going to compare the expression levels of all genes in two conditions, Ish and T47D.  We have 5 replicates for each condition.


Setup
=====

```{r message=FALSE}
library(tidyverse)
library(ggrepel)
```

Prepare the data
================

Read the TSV
------------

```{r "Read Data TSV", message=FALSE}
read_tsv(
  "rnaseq_counts.txt",
  guess_max=1000000
) -> data

head(data)
```
Restructuring
-------------

We're going to split apart the annotation and the data to save space.

```{r}
data %>%
  select(Probe:Distance) -> data.annotation
```

We can then remove the annotation from the data and restructure it.


```{r}
data %>%
  select(-(Chromosome:Distance)) %>%
  pivot_longer(
    cols=-Probe,
    names_to="sample",
    values_to="count"
  ) %>%
  separate(
    sample,
    into=c("condition","replicate"),
    sep="_",
    remove = FALSE
  ) -> data

head(data)
```
Let's remove any genes which are unmeasured in all samples.

```{r}
data %>%
  group_by(Probe) %>%
  summarise(
    max = max(count)
  ) %>%
  ungroup() %>%
  filter(max==0) %>%
  pull(Probe) -> unmeasured.probes


data %>%
  filter(! Probe %in% unmeasured.probes ) -> data
```

Normalisation
-------------

We want to express our data as log2RPM (reads per million reads of library)

```{r}
data %>%
  group_by(sample) %>%
  mutate(
    log2RPM = log2( (count+1) / (sum(count)/1000000) )
  ) %>%
  ungroup()-> data

head(data)
```

Normalisation check
-------------------

Let's check the distributions of all of the samples with a violin plot.

```{r}
data %>%
  ggplot(aes(x=sample, y=log2RPM, fill=condition)) +
  geom_violin() +
  scale_fill_brewer(palette = "Set1")

```

Average per gene
----------------

We want to know the mean expression per gene for each of the conditions.

```{r}
data %>%
  group_by(Probe, condition) %>%
  summarise(
    expression=mean(log2RPM)
  ) %>%
  ungroup() %>%
  pivot_wider(
    names_from=condition,
    values_from=expression
  ) -> data.per.sample

head(data.per.sample)
```

Plotting and Analysis
=====================

```{r fig.height=7, fig.width=7}
data.per.sample %>%
  ggplot(aes(x=Ish, y=T47D)) +
  geom_point(size=0.3)
```

Biggest Changes
---------------

```{r}
data.per.sample %>%
  mutate(diff=Ish-T47D) %>%
  arrange(desc(abs(diff))) -> data.per.sample

data.per.sample %>%
  slice(1:50)
```
GAGE Genes
----------

We heard that these were regulated together, so let's look.

```{r}
data.per.sample %>%
  filter(startsWith(Probe,"GAGE")) -> data.gage.genes

data.gage.genes
```
Plotting with a GAGE column
---------------------------

```{r fig.height=7, fig.width=7}
data.per.sample %>%
  mutate(
    IsGage = startsWith(Probe,"GAGE")
  ) %>%
  ggplot(aes(x=Ish,y=T47D,colour=IsGage)) +
  geom_point(size=0.3, show.legend = FALSE) +
  scale_colour_manual(values=c("grey","red2")) +
  theme_bw()
```


Plotting with a data subset
---------------------------

```{r fig.height=7, fig.width=7}
data.per.sample %>%
  ggplot(aes(x=Ish,y=T47D, label=Probe)) +
  geom_point(size=0.3, show.legend = FALSE, colour="grey") +
  theme_bw() +
  geom_point(
    size=2, 
    show.legend = FALSE, 
    colour="red2", 
    data=data.gage.genes
  ) +
  geom_text_repel(
    data=data.gage.genes,
    colour="black"
  )
```













